File: packages/e-top-horizon-plugin/lib/utils/transaction_type.dart
enum TransactionType {
  purchase,
  balance$Inquiry,
  reversal,
  refund,
  pre$Auth,
  cash$Advance,
  transaction$Completion,
  cash$Back
}

extension ParseToString on TransactionType {
  String toShortString() {
    return toString().split('.').last.replaceAll("\$", " ");
  }
}

File: packages/e-top-horizon-plugin/android/src/main/java/com/etop_pos_plugin/networking/model/TransactionType.java
package com.etop_pos_plugin.networking.model;

public enum TransactionType {
    PURCHASE("Purchase"),
    REVERSAL("Reversal"),
    REFUND("Refund"),
    BALANCE_INQUIRY("Balance Inquiry"),
    PRE_AUTH("Pre-Authorization"),
    PURCHASE_COMPLETION("Purchase Completion"),
    CASH_ADVANCE("Cash Advance"),
    CASH_BACK("Cash Back");

    private final String description;

    TransactionType(String description) {
        this.description = description;
    }

    public String getDescription() {
        return description;
    }

    @Override
    public String toString() {
        return description;
    }
}
